source/renderer/app/components/staking/epochs/StakingEpochs.js

```diff
@@ -1,15 +1,421 @@
+  },
+  tableHeaderPerformance: {
+    id: 'staking.epochs.tableHeader.performance',
+    defaultMessage: '!!!Performance',
+    description: 'Table header "Performance" label on staking epochs page',
+  },
+  tableHeaderSharedRewards: {
+    id: 'staking.epochs.tableHeader.sharedRewards',
+    defaultMessage: '!!!Shared rewards',
+    description: 'Table header "Shared rewards" label on staking epochs page',
+  },
+  tableBodySlots: {
+    id: 'staking.epochs.tableBody.slots',
+    defaultMessage: '!!!slots',
+    description: '"slots" text in table body on staking epochs page',
+  },
+  tableBodyOf: {
+    id: 'staking.epochs.tableBody.of',
+    defaultMessage: '!!!of',
+    description: '"of" text in table body on staking epochs page',
+  },
+  tableBodyAda: {
+    id: 'environment.currency.ada',
+    defaultMessage: '!!!Ada',
+    description: '"Ada" text in table body on staking epochs page',
+  },
+});
+
 type Props = {
-  name: string,
+  currentEpochName: string,
+  currentEpochData: any,
+  currentEpochEndDateTime: string,
+  currentEpochProgress: number,
+  previousEpochName: string,
+  previousEpochData: any,
+  isLoading: boolean,
+};
+
+type State = {
+  selectedEpoch: string,
+  currentEpochDataOrder: string,
+  previousEpochDataOrder: string,
+  currentEpochDataSortBy: string,
+  previousEpochDataSortBy: string,
 };
 
+const { CURRENT_EPOCH, PREVIOUS_EPOCH } = SELECTED_EPOCH_OPTIONS;
+
 @observer
-export default class StakingEpochs extends Component<Props> {
+export default class StakingEpochs extends Component<Props, State> {
+  static contextTypes = {
+    intl: intlShape.isRequired,
+  };
+
+  static defaultProps = {
+    isLoading: false,
+  };
+
+  constructor() {
+    super();
+    this.state = {
+      selectedEpoch: CURRENT_EPOCH,
+      currentEpochDataOrder: 'desc',
+      previousEpochDataOrder: 'desc',
+      currentEpochDataSortBy: 'pool',
+      previousEpochDataSortBy: 'pool',
+    };
+  }
+
+  onSelectedEpochChange = (selectedEpoch: string) =>
+    this.setState({ selectedEpoch });
+
+  handleDataSort = (selectedEpoch: string, newSortBy: string) => {
+    const {
+      currentEpochDataOrder,
+      currentEpochDataSortBy,
+      previousEpochDataOrder,
+      previousEpochDataSortBy,
+    } = this.state;
+    let newOrder;
+
+    if (selectedEpoch === CURRENT_EPOCH) {
+      if (currentEpochDataSortBy === newSortBy) {
+        newOrder = currentEpochDataOrder === 'asc' ? 'desc' : 'asc';
+      } else {
+        newOrder = 'desc';
+      }
+      this.setState({
+        currentEpochDataSortBy: newSortBy,
+        currentEpochDataOrder: newOrder,
+      });
+    } else {
+      if (previousEpochDataSortBy === newSortBy) {
+        newOrder = previousEpochDataOrder === 'asc' ? 'desc' : 'asc';
+      } else {
+        newOrder = 'desc';
+      }
+      this.setState({
+        previousEpochDataSortBy: newSortBy,
+        previousEpochDataOrder: newOrder,
+      });
+    }
+  };
+
+  renderNoData = () => {
```

Comment:
@yakovkaravelov please, turn all these rendering methods into separated components (e.g. `renderNoData` -> `<RenderNoData />`), as `StakingEpochs.js` is too large, which makes it harder to maintain:
* `renderNoData` -> <renderNoData />
* `renderDataTable`
* `renderCurrentEpochData`
* `renderPreviousEpochData`

Reply:
(no need to add `Render` to the names of the components, as my example wrongly did)

Reply:
@yakovkaravelov I have updated my comment, as I had wrongly said to name the component RenderNoData. Please, name them `StakingEpochs{xxx}.js`, instead